src/fireflyClient.ts

```diff
@@ -802,7 +804,7 @@ export class FireflyClient {
           leverage,
           this.maxBlockGasLimit,
           this.networkName,
-          this.getPublicAddress
+          parentAddress?()=>{return parentAddress}:this.getPublicAddress
```

Comment:
why not just do `parentAddress? parentAddress : this.getPublicAddress`

Reply:
also why this check is only in `else` clause, what if we are using biconomy and want to adjust parent's leverage?

Reply:
1) The function expects a callback , so either I can pass it like this or change the implementation of both the functions to get the address in string.

2) Yes good catch. It will also be there in  biconomy call.